src/Microsoft.NetCore.Analyzers/Core/Security/DoNotCallDangerousMethodsInDeserialization.cs

```diff
@@ -0,0 +1,231 @@
+                    {
+                        foreach (var targetMethodName in SystemIOFileMethodMetadataNames)
+                        {
+                            dangerousMethodSymbolsBuilder.AddRange(systemIOFileTypeSymbol.GetMembers(targetMethodName).Where(s => s is IMethodSymbol).Cast<IMethodSymbol>());
+                        }
+                    }
+
+                    var systemReflectionAssemblyTypeSymbol = WellKnownTypes.SystemReflectionAssembly(compilation);
+
+                    if (systemReflectionAssemblyTypeSymbol != null)
+                    {
+                        foreach (var targetMethodName in SystemReflectionAssemblyMethodMetadataNames)
+                        {
+                            dangerousMethodSymbolsBuilder.AddRange(systemReflectionAssemblyTypeSymbol.GetMembers(targetMethodName).Where(s => s is IMethodSymbol).Cast<IMethodSymbol>());
+                        }
+                    }
+
+                    var dangerousMethodSymbols = dangerousMethodSymbolsBuilder.ToImmutable();
+
+                    if (dangerousMethodSymbols.Length == 0)
+                    {
+                        return;
+                    }
+
+                    var attributeTypeSymbolsBuilder = ImmutableArray.CreateBuilder<INamedTypeSymbol>();
+                    var onDeserializingAttributeTypeSymbol = WellKnownTypes.OnDeserializingAttribute(compilation);
+
+                    if (onDeserializingAttributeTypeSymbol != null)
+                    {
+                        attributeTypeSymbolsBuilder.Add(onDeserializingAttributeTypeSymbol);
+                    }
+
+                    var onDeserializedAttributeTypeSymbol = WellKnownTypes.OnDeserializedAttribute(compilation);
+
+                    if (onDeserializedAttributeTypeSymbol != null)
+                    {
+                        attributeTypeSymbolsBuilder.Add(onDeserializedAttributeTypeSymbol);
+                    }
+
+                    var attributeTypeSymbols = attributeTypeSymbolsBuilder.ToImmutable();
+                    var visitedMethodSymbols = new HashSet<IMethodSymbol>();
+
+                    compilationStartAnalysisContext.RegisterSymbolAction(
+                        (SymbolAnalysisContext symbolAnalysisContext) =>
+                        {
+                            var classSymbol = symbolAnalysisContext.Symbol as INamedTypeSymbol;
+
+                            if (!classSymbol.GetAttributes().Any(s => s.AttributeClass.Equals(serializableAttributeTypeSymbol)))
+                            {
+                                return;
+                            }
+                            
+                            foreach (var member in classSymbol.GetMembers())
+                            {
+                                if (member.Kind == SymbolKind.Method)
+                                {
+                                    var methodSymbol = member as IMethodSymbol;
```

Comment:
I'd just write the loop as `foreach (var methodSymbol in classSymbol.GetMembers().OfType<IMethodSymbol>())`